plank/pkg/server/endpointer_handler_factory.go

```diff
@@ -0,0 +1,121 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/sirupsen/logrus"
+	"github.com/vmware/transport-go/bus"
+	"github.com/vmware/transport-go/model"
+	"github.com/vmware/transport-go/plank/utils"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// buildEndpointHandler builds a http.HandlerFunc that wraps Transport Bus operations in an HTTP request-response cycle.
+// service channel, request builder and rest bridge timeout are passed as parameters.
+func buildEndpointHandler(svcChannel string, reqBuilder func(w http.ResponseWriter, r *http.Request) model.Request, restBridgeTimeout time.Duration) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		defer func() {
+			if r := recover(); r != nil {
+				utils.Log.Errorln(r)
+				http.Error(w, "Internal Server Error", 500)
+			}
+		}()
+
+		// set context that expires after the provided amount of time in restBridgeTimeout to prevent requests from hanging forever
+		ctx, cancelFn := context.WithTimeout(context.Background(), restBridgeTimeout)
+		defer cancelFn()
+		h, err := bus.GetBus().ListenOnce(svcChannel)
+		if err != nil {
+			panic(err)
+		}
+
+		// set up a channel through which to receive the raw response from transport channel
+		// handler function runs in another thread so we need to utilize channel to use the correct writer.
+		chanReturn := make(chan *transportChannelResponse)
+		h.Handle(func(message *model.Message) {
+			chanReturn <- &transportChannelResponse{message: message}
+		}, func(err error) {
+			chanReturn <- &transportChannelResponse{err: err}
+		})
+
+		// relay the request to transport channel
+		reqModel := reqBuilder(w, r)
+		err = bus.GetBus().SendRequestMessage(svcChannel, reqModel, reqModel.Id)
+
+		// get a response from the channel, render the results using ResponseWriter and log the data/error
+		// to the console as well.
+		select {
+		case <-ctx.Done():
+			http.Error(w, "Request timed out", 500)
```

Comment:
It would be good to add a little more context here. `No response received from internal service in Nms, request timed out`